### PR TITLE
Improved/refactored bumping Gradle tasks

### PR DESCRIPTION
**Pull request by afalhambra-hivemq**
https://hivemq.kanbanize.com/ctrl_board/22/cards/19901/details/

Several improvements around these Gradle tasks:

1. Description, and show of the usage of each task, including their different parameters and which are optional.
2. No need to specify the `chart` version. When omitted, it will automatically bump the patch version of the chart (i.e. from `0.2.1` to `0.2.2`).
3. Added missing `example` to the ones for bumping the version.
4. Grouped three different Gradle tasks for bumping the version when releasing the platform, so only one single Gradle task is needed to be executed, called `updateAllPlatformChartVersions`.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -1,85 +1,128 @@
...
-    dependsOn(updateChartAndValueFilesWithVersion)
-    dependsOn(gradle.includedBuild("tests-hivemq-platform-operator").task(":updatePlatformVersion"))
-    doLast {
-        exec {
-            workingDir(projectDir)
-            commandLine("sh", "./manifests/hivemq-platform/manifests.sh")
-        }
+    description =
+        "Bumps Platform Chart and Platform versions. " +
+                "\n\tUsage: ./gradlew updatePlatformChartVersion -PchartVersion=x.y.z -PappVersion=x.y.z" +
```
Comment (ymengesha) [inline build.gradle.kts:63]: Nitpick:
```suggestion
                "\n\tUsage: ./gradlew updatePlatformChartVersion -PchartVersion=a.b.c -PappVersion=x.y.z" +
```

File: tests-hivemq-platform-operator/build.gradle.kts [hunk 2]
```diff
@@ -191,28 +193,32 @@ val updatePlatformVersion by tasks.registering {
...
-                val replacedText2 = replacedText1.replace("""^(hivemq\..*\.version)=(.*)$""".toRegex(RegexOption.MULTILINE)) { matchResult ->
-                    "${matchResult.groupValues[1]}=${appVersion}"
-                }
-                val replacedText3 = replacedText2.replace("""(?i)(hivemq/hivemq4:)(\d+\.\d+\.\d+(-snapshot)?)$""".toRegex(RegexOption.MULTILINE)) { matchResult ->
-                    "${matchResult.groupValues[1]}${appVersion}${matchResult.groupValues[3]}"
-                }
-                val replacedText4 = replacedText3.replace("""(?i)(hivemq/mqtt-cli:)(\d+\.\d+\.\d+(-snapshot)?)$""".toRegex(RegexOption.MULTILINE)) { matchResult ->
-                    "${matchResult.groupValues[1]}${appVersion}${matchResult.groupValues[3]}"
-                }
+                val replacedText1 =
```
Comment (ymengesha) [inline tests-hivemq-platform-operator/build.gradle.kts:206]: I think we can use one `replacedText` variable.

**Review by ymengesha — APPROVED** [review]
Just some very minor suggestions! Thank you for making these changes!

File: build.gradle.kts [hunk 3]
```diff
@@ -1,85 +1,128 @@
...
-    dependsOn(updateChartAndValueFilesWithVersion)
-    dependsOn(gradle.includedBuild("tests-hivemq-operator").task(":updatePlatformVersion"))
-    doLast {
-        exec {
-            workingDir(projectDir)
-            commandLine("sh", "./manifests/hivemq-operator/manifests.sh")
-        }
+    description =
+        "Bumps Operator Legacy Chart and Platform versions. " +
+                "\n\tUsage: ./gradlew updateOperatorChartVersion -PchartVersion=x.y.z -PappVersion=x.y.z" +
```
[the remark this answers is not in the document]
Comment (ymengesha) [reply build.gradle.kts:7]: This is just to hint that the versions are different.

**Comment by afalhambra-hivemq** [comment]
> Just some very minor suggestions! Thank you for making these changes!

Thanks for the review @ymengesha - addressed your comments.

**Review by Donnerbart — APPROVED** [review]
This is a great improvement!

**Pull request merged**